velox/expression/CastExpr.cpp

```diff
@@ -123,6 +123,37 @@ void applyIntToDecimalCastKernel(
   });
 }
 
+template <typename TInput>
+VectorPtr applyDecimalToIntCast(
+    const SelectivityVector& rows,
+    const BaseVector& input,
+    exec::EvalCtx& context,
+    const TypePtr& toType) {
+  VectorPtr result;
+  context.ensureWritable(rows, INTEGER(), result);
+  (*result).clearNulls(rows);
+  auto resultBuffer =
+      result->asUnchecked<FlatVector<int32_t>>()->mutableRawValues();
+  const auto precisionScale = getDecimalPrecisionScale(*toType);
+  const auto simpleInput = input.as<SimpleVector<TInput>>();
+  context.applyToSelectedNoThrow(rows, [&](int row) {
+    try {
+      int128_t integralPart = simpleInput->valueAt(row) /
+          DecimalUtil::kPowersOfTen[precisionScale.second];
+      resultBuffer[row] =
+          util::Converter<TypeKind::INTEGER, void, false>::cast(integralPart);
+    } catch (const VeloxRuntimeError& re) {
```

Comment:
Is this needed?

Reply:
It will throw out of range exceptions. Following patterns in applyCastPrimitivesDispatch.